dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsIT.java

```diff
@@ -0,0 +1,141 @@
+import io.dockstore.common.SourceControl;
+import io.dockstore.webservice.DockstoreWebserviceApplication;
+import io.dockstore.webservice.core.BioWorkflow;
+import io.dockstore.webservice.core.Partner;
+import io.dockstore.webservice.core.Workflow;
+import io.dockstore.webservice.core.WorkflowVersion;
+import io.dockstore.webservice.core.metrics.CpuStatisticMetric;
+import io.dockstore.webservice.core.metrics.ExecutionStatusCountMetric;
+import io.dockstore.webservice.core.metrics.ExecutionTimeStatisticMetric;
+import io.dockstore.webservice.core.metrics.MemoryStatisticMetric;
+import io.dockstore.webservice.core.metrics.Metrics;
+import io.dockstore.webservice.jdbi.MetricsDAO;
+import io.dockstore.webservice.jdbi.WorkflowDAO;
+import io.dockstore.webservice.jdbi.WorkflowVersionDAO;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.context.internal.ManagedSessionContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class MetricsIT extends BaseIT {
+
+    private MetricsDAO metricsDAO;
+    private Session session;
+    private WorkflowDAO workflowDAO;
+    private WorkflowVersionDAO workflowVersionDAO;
+
+    @BeforeEach
+    public void setup() {
+        DockstoreWebserviceApplication application = SUPPORT.getApplication();
+        SessionFactory sessionFactory = application.getHibernate().getSessionFactory();
+        this.metricsDAO = new MetricsDAO(sessionFactory);
+        this.workflowDAO = new WorkflowDAO(sessionFactory);
+        this.workflowVersionDAO = new WorkflowVersionDAO(sessionFactory);
+
+        // used to allow us to use DAOs outside of the web service
+        this.session = application.getHibernate().getSessionFactory().openSession();
+        ManagedSessionContext.bind(session);
+    }
+
+    @BeforeEach
+    @Override
+    public void resetDBBetweenTests() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false, testingPostgres);
+    }
+
+    /**
+     * Demos the creation, update, and persistence of the Metrics entity
+     */
+    @Test
+    void testDAOs() {
+        final Transaction transaction = session.beginTransaction();
+
+        Metrics metrics1 = createMetrics();
+        Metrics metrics2 = createMetrics();
+        metricsDAO.create(metrics1);
+        metricsDAO.create(metrics2);
+
+        // Create a workflow and workflow version so we can add metrics for a specific platform to the workflow version
+        Workflow workflow = createWorkflow();
+        workflowDAO.create(workflow);
+        WorkflowVersion workflowVersion = createWorkflowVersion(workflow);
+        workflowVersion.getMetricsByPlatform().put(Partner.TERRA, metrics1);
+        workflowVersion.getMetricsByPlatform().put(Partner.DNA_STACK, metrics2);
+        workflowVersionDAO.create(workflowVersion);
+
+        session.flush();
+        transaction.commit();
+        session.close();
```

Comment:
would also like to see an example of updating the statistics and then seeing the average, min, maximums change
(this unlike the comment just above can be a follow-up PR)

Reply:
https://ucsc-cgl.atlassian.net/browse/SEAB-5063

Reply:
I think it makes the most sense to do it in one of the metrics endpoints ticket (like the one you linked) where the full life cycle can be used. There's no way to add one data point in the webservice right now since the data will be stored in S3. When it's ready, I'll make sure to have a test that:
- aggregates existing data
- adds more data points to S3
- re-aggregates so that we can verify the updated aggregated metrics